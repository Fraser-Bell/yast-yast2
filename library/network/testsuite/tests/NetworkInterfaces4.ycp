{
// bug 72164

include "testsuite.ycp";

map READ = $[
    "network" : $[
	"section" : $[
	    "eth4"	: nil,
	],
	"value" : $[
	    "eth4"	: $["BOOTPROTO":"dhcp", "NAME": "single quotes aren't good"],
	]
    ],
    "probe" : $[
	"system" : [],
    ],
    "target" : $[
	"tmpdir" : "/tmp",
    ],
];

map EXEC = $[
    "target" : $[
        "bash_output": $[
                "exit": 0,
                "stdout": "",
                "stderr":"",
        ],
    ],
];

TESTSUITE_INIT([READ, $[], EXEC], nil);
import "NetworkInterfaces";

DUMP("NetworkInterfaces::Read");
TEST(``(NetworkInterfaces::Read()), [READ, $[], EXEC], nil);
NetworkInterfaces::OriginalDevices = nil;

DUMP("NetworkInterfaces::Write");
TEST(``(NetworkInterfaces::Write("")), [READ], nil);
}
