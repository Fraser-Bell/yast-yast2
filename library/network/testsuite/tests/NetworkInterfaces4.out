Read	.target.tmpdir "/tmp"
Dump	NetworkInterfaces::Read
Dir	.network.section: ["eth4"]
Dir	.network.value."eth4": ["BOOTPROTO", "NAME"]
Read	.network.value."eth4".BOOTPROTO "dhcp"
Read	.network.value."eth4".NAME "single quotes aren't good"
Read	.target.stat "/sys/class/net/eth4/type" nil
Read	.target.string "/sys/class/net/eth4/type" nil
Return	true
Dump	NetworkInterfaces::Write
Write	.network.value."eth4".BOOTPROTO "dhcp" true
Write	.network.value."eth4".NAME "single quotes arent good" true
Write	.network.value."eth4".STARTMODE "" true
Write	.network nil true
Return	true
